### PR TITLE
Add pre-validation checks to CS upgrade scripts

### PR DESCRIPTION
**Pull request by siyer-123**
- Added pre-validation checks to confirm upgrade status prior to running script (ported from Common Services version of script)
- Added airgap version of the script that the CS team made (also includes the pre-validation checks mentioned above)

### CODE REVIEW

File: upgrade/upgrade_common_services.sh [hunk 1]
```diff
@@ -107,6 +107,8 @@ function check_preqreqs() {
     local csNS=$1
     local cloudpaksNS=$2
     local controlNS=$3
+
+    msg ""
```
Comment (tseelbach) [inline upgrade/upgrade_common_services.sh:111]: prereqs is spelled wrong on 112

File: upgrade/upgrade_common_services.sh [hunk 2]
```diff
@@ -185,6 +187,46 @@ EOF
     fi
 }
 
+function compare_channel() {
```
Comment (tseelbach) [inline upgrade/upgrade_common_services.sh:190]: Please add a comment, sentence or 2 about the intent of this function. I can see its comparing channels, but why? Why is this function needed now when we didn't need it before 3.5?

File: upgrade/upgrade_common_services.sh [hunk 3]
```diff
@@ -195,20 +237,52 @@ function switch_channel() {
 
     STEP=$((STEP + 1 ))
 
-    title "[${STEP}] Switching channel into ${channel}..."
+    msg ""
+    title "[${STEP}] Compareing given upgrade channel version ${channel} with current one ..."
```
Comment (tseelbach) [inline upgrade/upgrade_common_services.sh:241]: comparing spelling

File: upgrade/upgrade_common_services_airgap.sh [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+    local subName=$1
+    local csNS=$2
+    local cloudpaksNS=$3
+    local controlNS=$4
+    local channel=$5
+    local allNamespace=$6
+
+    STEP=$((STEP + 1 ))
+    msg ""
+    title "[${STEP}] Compareing given upgrade channel version ${channel} with current one ..."
```
Comment (tseelbach) [inline upgrade/upgrade_common_services_airgap.sh:271]: comparing spelling

File: upgrade/upgrade_common_services_airgap.sh [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+    subs=$1
+    ns=$2
+    for sub in ${subs}; do
+        exist=$(oc get sub ${sub} -n ${ns} --ignore-not-found)
+        if [[ "X${exist}" != "X" ]]; then
+            msg "Deleting ${sub} in namesapce ${ns}, it will be re-installed after the upgrade is successful ..."
+            msg "-----------------------------------------------------------------------"
+            csv=$(oc get sub ${sub} -n ${ns} -o=jsonpath='{.status.installedCSV}' --ignore-not-found)
+            in_step=1
+            msg "[${in_step}] Removing the subscription of ${sub} in namesapce ${ns} ..."
```
Comment (tseelbach) [inline upgrade/upgrade_common_services_airgap.sh:373]: namespace spelling

File: upgrade/upgrade_common_services_airgap.sh [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+            oc delete sub ${sub} -n ${ns} --ignore-not-found
+            in_step=$((in_step + 1))
+            msg "[${in_step}] Removing the csv of ${sub} in namesapce ${ns} ..."
```
Comment (tseelbach) [inline upgrade/upgrade_common_services_airgap.sh:376]: namespace spelling

**Review by tseelbach — CHANGES_REQUESTED** [review]
we should fix the spelling errors locally and then open an issue on bedrock to make same changed. It may seem petty but if you spend $ millions on IBM software and you bet your business on it, we need to at least have correct spelling in the messages.  It helps to instill confidence in the SysAdmins that have to run these processes

**Comment by siyer-123** [comment]
Resolved all the comments above. I will open up a roadmap issue for the Common Services version of the script to get those spelling mistakes resolved at the source.

**Pull request merged**